### PR TITLE
Bluetooth: controller: LLCP: Rejection logic for handling non-supported features

### PR DESCRIPTION
**Pull request by thoh-ot**
Implement rejection logic for handling non-supported features and unit tests for:

- Invalid opcodes based on controller role
- No LE Encryption support
- No Connection Parameters Request procedure support
- No Peripheral-initiated Features Exchange support
- No LE 2M PHY/LE Coded PHY support

Update ztest for non-kernel test to support skipping via `ztest_test_skip()`.

### CODE REVIEW

**Review by erbr-ot — COMMENTED** [review]
Looks real good to me. Look forward to trying it and perhaps to get a walk through on Thursday?

File: subsys/bluetooth/controller/ll_sw/ull_llcp_remote.c [hunk 1]
```diff
@@ -383,6 +392,25 @@ static void rr_act_reject(struct ll_conn *conn)
...
+static void rr_act_unsupported(struct ll_conn *conn)
+{
+	struct proc_ctx *ctx = llcp_rr_peek(conn);
+
+	LL_ASSERT(ctx != NULL);
+
+	if (ctx->pause || !llcp_tx_alloc_peek(conn, ctx)) {
+		rr_set_state(conn, RR_STATE_UNSUPPORTED);
+	} else {
+		struct proc_ctx *ctx = llcp_rr_peek(conn);
```
Comment (kruithofa) [inline subsys/bluetooth/controller/ll_sw/ull_llcp_remote.c:404]: This is a merge error and needs to be fixed

File: tests/bluetooth/controller/ctrl_unsupported/src/main.c [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+
+	/* There should not be a host notifications */
+	ut_rx_q_is_empty();
+
+	zassert_equal(ctx_buffers_free(), CONFIG_BT_CTLR_LLCP_PROC_CTX_BUF_NUM,
+		      "Free CTX buffers %d", ctx_buffers_free());
+}
+
+void test_invalid_per_rem(void)
+{
```
Comment (kruithofa) [inline tests/bluetooth/controller/ctrl_unsupported/src/main.c:106]: add test for undefined opcode

Comment (kruithofa) [reply tests/bluetooth/controller/ctrl_unsupported/src/main.c:106]: in particular opcode 255 may be an issue

**Review by kruithofa — DISMISSED** [review]

**Review by erbr-ot — DISMISSED** [review]
LGTM

File: tests/bluetooth/controller/ctrl_unsupported/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,17 @@
+# SPDX-License-Identifier: Apache-2.0
+
+cmake_minimum_required(VERSION 3.13.1)
```
Comment (tejlmand) [inline tests/bluetooth/controller/ctrl_unsupported/CMakeLists.txt:3]:
```suggestion
cmake_minimum_required(VERSION 3.20)
```

File: tests/bluetooth/controller/ctrl_unsupported/testcase.yaml [hunk 4]
```diff
@@ -3,3 +3,15 @@ common:
 tests:
   bluetooth.controller.ctrl_unsupported.default.test:
      type: unit
+  bluetooth.controller.ctrl_unsupported.no_enc.test:
+     type: unit
+     extra_args: NO_ENC=y KCONFIG_OVERRIDE_FILE="kconfig_override.h"
```
Comment (tejlmand) [inline tests/bluetooth/controller/ctrl_unsupported/testcase.yaml:8]: seems like I was not invited to this party #40073

The way `KCONFIG_OVERRIDE_FILE` and `tests/bluetooth/controller/common/defaults_cmake.txt` appears to be short cutting everything else we do in the build system.

But guess that is outside the scope of this PR.

**Review by tejlmand — DISMISSED** [review]
CMake version is now 3.20

**Review by ppryga-nordic — APPROVED** [review]
I did quick pass-through and I didn't find anything blocking the PR so approve it.

**Review by kruithofa — APPROVED** [review]

**Pull request merged**